tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaConsumerConstructorIntegration.cs

```diff
@@ -50,9 +65,9 @@ internal static CallTargetReturn OnMethodEnd<TTarget>(TTarget instance, Exceptio
     {
         // This method is called in the Consumer constructor, so if we have an exception
         // the consumer won't be created, so no point recording it.
-        if (exception is null && state.State is string s)
+        if (exception is not null)
         {
-            ConsumerGroupHelper.SetConsumerGroup(instance, s);
+            ConsumerGroupHelper.RemoveConsumerGroup(instance);
         }
```

Comment:
This will now call `ConsumerGroupHelper.RemoveConsumerGroup` when the integration is disabled. Presumably not a big deal, and I think avoiding it would require either checking _again_ whether enabled (doing more work than necessary), or passing a flag in `State` to indicate that it's enabled (which would be boxed). So just pointing it out rather than giving a solution :D

Reply:
Done in 036435648dcab79ccb1a05690099311a8020890c